enumerable.rb

```diff
@@ -0,0 +1,124 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_map(&block)
+    result = []
+    each do |element|
+      result <<
+        block.call(element)
+    end
+    result
+  end
+
+  def my_select(&block)
+    result = []
+    each do |element|
+      result << element if block.call(element) == true
+    end
+    result
+  end
+
+  def my_each
+    i = 0
+    while i < size
+      yield self[i]
+      i += 1
+    end
+    i
+  end
+
+  def my_each_with_index
+    num = 0
+    each do |words|
+      yield(words, num)
+      num += 1
+    end
+  end
+
+  def my_any?(*args)
+    if !args[0].nil?
+      my_each { |item| return false if args[0] == item }
+    elsif block_given?
+      my_each { |item| return true if item }
+    elsif args.instance_of?(Class)
+      my_each { |_item| return true if items.instance_of?(args) == args }
+    elsif args.instance_of?(Regexp)
+      my_each { |item| return true if args.match(item) }
+    else
+      my_each { |item| return true if item == args }
+    end
+    true
+  end
```

Comment:
- [ ]  This method returns true for all test cases, kindly fix this
![Screenshot (305)](https://user-images.githubusercontent.com/86371332/140478950-21cb3452-991d-4214-8590-2284fbf188d7.png)